### PR TITLE
Add `BookmarkList` example

### PR DESCRIPTION
**Pull request by vanillajonathan**
Add example for #1346.

### CODE REVIEW

**Comment by bilelmoussaoui** [comment]
I was expecting a proper example in https://github.com/gtk-rs/gtk4-rs/tree/master/examples instead

**Comment by vanillajonathan** [comment]
Oh yeah, that is a good idea too!

I was thinking of having a example  section in the docblock to sit shows in the documentation and when using the "Go to definition" in the IDE.

Does this project accept examples in the docblock too?

File: gtk4/src/bookmark_list.rs [hunk 1]
```diff
@@ -3,6 +3,16 @@
 use crate::BookmarkList;
 use glib::translate::*;
 
+/// # Example
+/// To iterate over `BookmarkList` call `iter::<gio::FileInfo>` to get an iterator.
+/// ```no_run
+/// let bookmarks = gtk::BookmarkList::new(Some("bookmarks.xbel"), None);
+/// for bookmark in bookmarks.iter::<gio::FileInfo>() {
```
[the remark this answers is not in the document]
Comment (vanillajonathan) [reply gtk4/src/bookmark_list.rs:10]: @bilelmoussaoui What do you think?

[the remark this answers is not in the document]
Comment (bilelmoussaoui) [reply gtk4/src/bookmark_list.rs:10]: I agree with what @zekefast said. Although, does this example shows in the docs?

[the remark this answers is not in the document]
Comment (vanillajonathan) [reply gtk4/src/bookmark_list.rs]: Okay I updated the PR with the example provided by @zekefast. I am not sure, but it should show up when you press <kbd>F12</kbd> in VS Code.

[the remark this answers is not in the document]
Comment (bilelmoussaoui) [reply gtk4/src/bookmark_list.rs]: You would have to resolve merge conflicts, the PR as is can't be tested by the CI nor merged

**Comment by bilelmoussaoui** [comment]
Please test locally first....

**Pull request closed without merging**